pkgs/tools/video/blackmagic-desktop-video/default.nix

```diff
@@ -0,0 +1,118 @@
+{ stdenv
+, cacert
+, curl
+, runCommandLocal
+, lib
+, autoPatchelfHook
+, libcxx
+, libcxxabi
+, libGL
+, gcc7
+}:
+
+stdenv.mkDerivation rec {
+  pname = "blackmagic-desktop-video";
+  version = "12.5a15";
+
+  buildInputs = [
+    autoPatchelfHook
+    libcxx
+    libcxxabi
+    libGL
+    gcc7.cc.lib
+  ];
+
+  # yes, the below download function is an absolute mess.
+  # blame blackmagicdesign.
+  src = runCommandLocal "${pname}-${lib.versions.majorMinor version}-src.tar.gz"
+    rec {
+      outputHashMode = "recursive";
+      outputHashAlgo = "sha256";
+      outputHash = "sha256-ss7Ab5dy7cmXp9LBirFXMeGY4ZbYHvWnXmYvNeBq0RY=";
+
+      impureEnvVars = lib.fetchers.proxyImpureEnvVars;
+
+      nativeBuildInputs = [ curl ];
+
+      # ENV VARS
+      SSL_CERT_FILE = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+
+      # from the URL that the POST happens to, see browser console
+      DOWNLOADID = "fecacc0f9b2f4c2e8bf2863e9e26c8e1";
+      # from the URL the download page where you click the "only download" button is at
+      REFERID = "052d944af6744608b27da496dfc4396d";
+      SITEURL = "https://www.blackmagicdesign.com/api/register/us/download/${DOWNLOADID}";
+
+      USERAGENT = builtins.concatStringsSep " " [
+        "User-Agent: Mozilla/5.0 (X11; Linux ${stdenv.targetPlatform.linuxArch})"
+        "AppleWebKit/537.36 (KHTML, like Gecko)"
+        "Chrome/77.0.3865.75"
+        "Safari/537.36"
+      ];
+
+      REQJSON = builtins.toJSON {
+        "country" = "nl";
+        "downloadOnly" = true;
+        "platform" = "Linux";
+        "policy" = true;
+      };
+
+    } ''
+    RESOLVEURL=$(curl \
+      -s \
+      -H 'Host: www.blackmagicdesign.com' \
+      -H 'Accept: application/json, text/plain, */*' \
+      -H 'Origin: https://www.blackmagicdesign.com' \
+      -H "$USERAGENT" \
+      -H 'Content-Type: application/json;charset=UTF-8' \
+      -H "Referer: https://www.blackmagicdesign.com/support/download/$REFERID/Linux" \
+      -H 'Accept-Encoding: gzip, deflate, br' \
+      -H 'Accept-Language: en-US,en;q=0.9' \
+      -H 'Authority: www.blackmagicdesign.com' \
+      -H 'Cookie: _ga=GA1.2.1849503966.1518103294; _gid=GA1.2.953840595.1518103294' \
+      --data-ascii "$REQJSON" \
+      --compressed \
+      "$SITEURL")
+
+    curl \
+      --retry 3 --retry-delay 3 \
+      -H "Host: swr.cloud.blackmagicdesign.com" \
+      -H "Upgrade-Insecure-Requests: 1" \
+      -H "$USERAGENT" \
+      -H "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8" \
+      -H "Accept-Language: en-US,en;q=0.9" \
+      --compressed \
+      "$RESOLVEURL" \
+      > $out
+  '';
+
+  setSourceRoot = ''
+    tar xf Blackmagic_Desktop_Video_Linux_${lib.versions.majorMinor version}/other/x86_64/desktopvideo-${version}-x86_64.tar.gz
+    sourceRoot=$NIX_BUILD_TOP/desktopvideo-${version}-x86_64
+  '';
```

Comment:
since this and a bunch of other stuff is used in both packages maybe split it out in to another generic.nix file like for example woodpecker.

Reply:
Hmm, unsure whether that is the right thing to do. Woodpecker is all normal packages; whereas the decklink driver consists of a kernel module and a support application (well, technically some desktop applications as well, but I have yet to get them running on NixOS and they're not required to get the capture card working).

Reply:
The two packages just inherently belong into two very different parts of the source tree; and I believe that `inherit`ing some parts like I have done for the kernel module is the best option, but I also think that behaviour like that should be as limited as possible.